korge/src/jvmMain/kotlin/com/soywiz/korge/testing/testing_utils.kt

```diff
@@ -0,0 +1,339 @@
+        val fileName = makeGoldenFileNameWithExtension(testMethodName, goldenName)
+        tempVfs[fileName].writeBitmap(bitmap, PNG)
+    }
+
+    suspend fun processGoldenResults() {
+        println("Processing golden results")
+        recordedGoldenNames.forEach { goldenName ->
+            println("Processing: $goldenName")
+            val goldenFileName = makeGoldenFileNameWithExtension(testMethodName, goldenName)
+            val testGoldenImage = testGoldensVfs[goldenFileName]
+            val existsGoldenInGoldenDirectory = testGoldenImage.exists()
+            if (existsGoldenInGoldenDirectory) {
+                val oldBitmap = testGoldenImage.readBitmap(PNG)
+                val newBitmap = tempVfs[goldenFileName].readBitmap(PNG)
+                if (!oldBitmap.contentEquals(newBitmap)) {
+                    testResultsOutput.results.add(
+                        KorgeTestResult.Diff(
+                            testMethodName,
+                            goldenName,
+                            testGoldensVfs,
+                            tempVfs,
+                            oldBitmap,
+                            newBitmap
+                        )
+                    )
+                }
+            } else {
+                // This is a new golden, so we can just save it.
+                tempVfs[goldenFileName].copyTo(testGoldenImage)
+            }
+        }
+
+        // Process any deleted goldens.
+        (existingGoldenNames - recordedGoldenNames).forEach { goldenName ->
+            val goldenFileName = makeGoldenFileNameWithExtension(testMethodName, goldenName)
+            val oldBitmap = testGoldensVfs[goldenFileName].readBitmap(PNG)
+            testResultsOutput.results.add(
+                KorgeTestResult.Deleted(
+                    testMethodName,
+                    goldenName,
+                    testGoldensVfs,
+                    oldBitmap
+                )
+            )
+        }
+    }
+
+    suspend fun endTest() {
+        println("Ending test")
+        processGoldenResults()
+
+        views.gameWindow.exitProcessOnExit = false
+        views.gameWindow.close()
+        testingLock.unlock()
+    }
+
+    companion object {
+        val DATE_FORMAT = DateFormat("yyyy-dd-MM HH:mm:ss z")
+        val PATH_DATE_FORMAT = DateFormat("yyyyMMdd_HH_mm_ss_z")
+        private val LINE_BREAK_WIDTH = 100
+    }
+}
+
+@OptIn(KorgeExperimental::class)
+inline fun Any.korgeTest(
+    korgeConfig: Korge.Config,
+    crossinline callback: suspend Stage.(korgeTester: KorgeTester) -> Unit,
+) {
+    System.setProperty("java.awt.headless", "false")
+    val throwable = Throwable()
+    val testClassName = throwable.stackTrace[0].className
+    val testMethodName = throwable.stackTrace[0].methodName
+    val existingMain = korgeConfig.main
+    val results = KorgeTestResults(testMethodName)
+    // Keep locked while tester is running
+    val testingLock = Mutex(locked = true)
+    korgeConfig.main = {
```

Comment:
To avoid making the configuration mutable. You can do something like:

```kotlin
val finalKorgeConfig = korgeConfig.copy(main = {
 ...

Then:

```kotlin
Korge(finalKorgeConfig)
```
})
```

Reply:
done